upup/pkg/fi/cloudup/openstacktasks/instance.go

```diff
@@ -151,12 +179,23 @@ func (_ *Instance) ShouldCreate(a, e, changes *Instance) (bool, error) {
 	return a == nil, nil
 }
 
+func makeServerName(e *Instance) string {
+	hash := utils.RandomString(6)
+	// FIXME: Must ensure 63 or less characters
+	// replace all dots and _ with -, this is needed to get external cloudprovider working
+	stripped := strings.Replace(fi.StringValue(e.ClusterName), ".k8s.local", "", -1)
+	iName := strings.Replace(strings.ToLower(fmt.Sprintf("%s-%s.%s", fi.StringValue(e.GroupName), hash, stripped)), "_", "-", -1)
+	instanceName := strings.Replace(iName, ".", "-", -1)
+	return instanceName
+}
+
 func (_ *Instance) RenderOpenstack(t *openstack.OpenstackAPITarget, a, e, changes *Instance) error {
 	if a == nil {
-		klog.V(2).Infof("Creating Instance with name: %q", fi.StringValue(e.Name))
-
+		e.Metadata[openstack.TagKopsName] = fi.StringValue(e.Name)
```

Comment:
What is the difference between `e.Name` and the result of `makeServerName(e)`?

Reply:
e.Name is the old stylish name for the instance. makeServerName returns the new style name. Which contains hash instead of index.

Reply:
So, in the kops world we still work with the old names whereas in Openstack we use the new ones? I tend to like it more to be consistent between those two worlds. Is that possible?